tests/unit/test_data.py

```diff
@@ -61,31 +60,14 @@ def test_dataset_raises_for_different_leading_shapes(
         Dataset(query_points, observations)
 
 
-@pytest.mark.parametrize(
-    "query_points_shape, observations_shape",
-    [
-        ((1, 2), (1,)),
-        ((1, 2), (1, 2, 3)),
-    ],
-)
-def test_dataset_raises_for_different_ranks(
```

Comment:
Maybe update test_dataset_concatenation with an example that has different ranks (but the same leading dimension)?